lbrynet/blob/blob_file.py

```diff
@@ -0,0 +1,227 @@
+            a instance of itself as HashBlob
+        """
+        if not peer in self.writers:
+            log.debug("Opening %s to be written by %s", str(self), str(peer))
+            finished_deferred = defer.Deferred()
+            writer = HashBlobWriter(self.get_length, self.writer_finished)
+            self.writers[peer] = (writer, finished_deferred)
+            return (writer, finished_deferred)
+        log.warning("Tried to download the same file twice simultaneously from the same peer")
+        return None, None
+
+    def open_for_reading(self):
+        """
+        open blob for reading
+
+        returns a file handle that can be read() from.
+        once finished with the file handle, user must call close_read_handle()
+        otherwise blob cannot be deleted.
+        """
+        if self._verified is True:
+            file_handle = None
+            try:
+                file_handle = open(self.file_path, 'rb')
+                self.readers += 1
+                return file_handle
+            except IOError:
+                log.exception('Failed to open %s', self.file_path)
+                self.close_read_handle(file_handle)
+        return None
+
+    def delete(self):
+        """
+        delete blob file from file system, prevent deletion
+        if a blob is being read from or written to
+
+        returns a deferred that firesback when delete is completed
+        """
+        if not self.writers and not self.readers:
+            self._verified = False
+            self.moved_verified_blob = False
+
+            def delete_from_file_system():
+                if os.path.isfile(self.file_path):
+                    os.remove(self.file_path)
+
+            d = threads.deferToThread(delete_from_file_system)
+
+            def log_error(err):
+                log.warning("An error occurred deleting %s: %s",
+                            str(self.file_path), err.getErrorMessage())
+                return err
+
+            d.addErrback(log_error)
+            return d
+        else:
+            return defer.fail(Failure(
+                ValueError("File is currently being read or written and cannot be deleted")))
+
+    @property
+    def verified(self):
+        """
+        Protect verified from being modified by other classes.
+        verified is True if a write to a blob has completed succesfully,
+        or a blob has been read to have the same length as specified
+        in init
+        """
+        return self._verified
+
+    def set_length(self, length):
+        if self.length is not None and length == self.length:
+            return True
+        if self.length is None and 0 <= length <= conf.settings['BLOB_SIZE']:
+            self.length = length
+            return True
+        log.warning("Got an invalid length. Previous length: %s, Invalid length: %s",
+                    self.length, length)
+        return False
+
+    def get_length(self):
```

Comment:
It's because a length getter is required rather than the current value.

https://github.com/lbryio/lbry/blob/dcad558259fff5574c361b20ce292da6820f3a32/lbrynet/blob/writer.py#L30